sql/src/main/java/io/crate/lucene/AbstractAnyQuery.java

```diff
@@ -36,13 +36,27 @@
 
 import javax.annotation.Nullable;
 import java.io.IOException;
+import java.util.List;
 
 abstract class AbstractAnyQuery implements FunctionToQuery {
 
+    private boolean ignoreCase;
+
+    boolean ignoreCase() {
+        return ignoreCase;
+    }
+
     @Override
     public Query apply(Function function, LuceneQueryBuilder.Context context) throws IOException {
-        Symbol left = function.arguments().get(0);
-        Symbol collectionSymbol = function.arguments().get(1);
+        List<Symbol> arguments = function.arguments();
+        Symbol left = arguments.get(0);
+        Symbol collectionSymbol = arguments.get(1);
+        if (arguments.size() == 3) {
+            Symbol ignoreCaseSymbol = arguments.get(2);
+            if (ignoreCaseSymbol instanceof Literal) {
+                ignoreCase = ((Literal<Boolean>) ignoreCaseSymbol).value();
```

Comment:
Same here.. that `AbstractAnyQuery` becomes coupled to implementation details of `Like` is not very nice, and that it is doing it on the basis of the number of arguments looks fragile to me. 

I think we should try a variant where we have a dedicated `ilike` function.